gtest/src/wasm_executor.rs

```diff
@@ -0,0 +1,352 @@
+            system.0.borrow_mut().fetch_inc_message_nonce().into(),
+            user_id,
+            Option::<Id>::encode(&Some(Id {
+                decimal: 2,
+                hex: vec![2u8],
+            })),
+            0,
+            0,
+            None,
+        );
+
+        let result = system.0.borrow_mut().call_meta(
+            user_id,
+            &program.id,
+            Some(meta_state_message),
+            meta_state_function_name,
+        );
+
+        assert_eq!(result, vec![0]);
+    }
+
+    #[test]
+    fn test_manager_executions_coworking() {
+        let meta_state_function_name = "meta_state";
+        let user_id: ProgramId = 100.into();
+        let system = System::default();
+        let program = Program::from_file(
+            &system,
+            "../target/wasm32-unknown-unknown/release/demo_meta.wasm",
+        );
+
+        let expected_result = vec![Wallet {
+            id: Id {
+                decimal: 2,
+                hex: vec![2u8],
+            },
+            person: Person {
+                surname: "OtherName".into(),
+                name: "OtherSurname".into(),
+            },
+        }];
+
+        let expected_id = Some(expected_result.first().unwrap().id.clone());
+
+        let meta_state_message = IncomingMessage::new(
+            system.0.borrow_mut().fetch_inc_message_nonce().into(),
+            user_id,
+            Option::<Id>::encode(&expected_id),
+            0,
+            0,
+            None,
+        );
+
+        let run_result = program.send(
+            user_id,
+            MessageInitIn {
+                amount: 1,
+                currency: "1".to_string(),
+            },
+        );
+        assert!(!run_result.main_failed);
+        let result = system.0.borrow_mut().call_meta(
+            user_id,
+            &program.id,
+            Some(meta_state_message),
+            meta_state_function_name,
+        );
+
+        assert_eq!(result, expected_result.encode());
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Source id in message is not equal to source id in function arguments"
+    )]
+    fn fool_test_for_correct_input() {
+        let meta_state_function_name = "meta_state";
+        let system = System::default();
+        let program = Program::from_file(
+            &system,
+            "../target/wasm32-unknown-unknown/release/demo_meta.wasm",
+        );
+
+        let meta_state_message = IncomingMessage::new(
+            system.0.borrow_mut().fetch_inc_message_nonce().into(),
+            100.into(),
+            Option::<Id>::encode(&Some(Id {
+                decimal: 2,
+                hex: vec![2u8],
+            })),
+            0,
+            0,
+            None,
+        );
+
+        system.0.borrow_mut().call_meta(
+            200.into(),
+            &program.id,
+            Some(meta_state_message),
+            meta_state_function_name,
+        );
```

Comment:
this should be also available directly from `Program`. Especially with `meta_state` only function

like `program.call_meta(arg)`